pkg/util/openstack/loadbalancer.go

```diff
@@ -257,6 +257,42 @@ func GetPoolByName(client *gophercloud.ServiceClient, name string, lbID string)
 	return &listenerPools[0], nil
 }
 
+// GetPoolsByListener finds pool for a listener. A listener always has exactly one pool.
+func GetPoolByListener(client *gophercloud.ServiceClient, lbID, listenerID string) (*pools.Pool, error) {
+	listenerPools := make([]pools.Pool, 0, 1)
+	err := pools.List(client, pools.ListOpts{LoadbalancerID: lbID}).EachPage(func(page pagination.Page) (bool, error) {
+		poolsList, err := pools.ExtractPools(page)
+		if err != nil {
+			return false, err
+		}
+		for _, p := range poolsList {
+			for _, l := range p.Listeners {
+				if l.ID == listenerID {
+					listenerPools = append(listenerPools, p)
+				}
+			}
+		}
+		if len(listenerPools) > 1 {
+			return false, ErrMultipleResults
+		}
+		return true, nil
+	})
+	if err != nil {
+		if cpoerrors.IsNotFound(err) {
+			return nil, ErrNotFound
+		}
+		return nil, err
+	}
+
+	if len(listenerPools) == 0 {
+		return nil, ErrNotFound
+	} else if len(listenerPools) > 1 {
```

Comment:
Is this check required here again? with L275 , if its multiple results, already might have handled by L280 ?

Reply:
This is copy from the old version, but I can fix this.